packages/guides/src/ReferenceResolvers/PageHyperlinkResolver.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\ReferenceResolvers;
+
+use phpDocumentor\Guides\Nodes\Inline\HyperLinkNode;
+use phpDocumentor\Guides\Nodes\Inline\LinkInlineNode;
+use phpDocumentor\Guides\RenderContext;
+use phpDocumentor\Guides\Renderer\UrlGenerator\UrlGeneratorInterface;
+
+use function rtrim;
+
+/**
+ * Resolves named and anonymous references to source files
+ *
+ * `Link Text <../page.rst>`_ or [Link Text](path/to/another/page.md)
+ */
+class PageHyperlinkResolver implements ReferenceResolver
+{
+    // Named links and anchors take precedence
+    public final const PRIORITY = -200;
+
+    public function __construct(
+        private readonly UrlGeneratorInterface $urlGenerator,
+        private readonly DocumentNameResolverInterface $documentNameResolver,
+    ) {
+    }
+
+    public function resolve(LinkInlineNode $node, RenderContext $renderContext): bool
+    {
+        if (!$node instanceof HyperLinkNode) {
+            return false;
+        }
+
+        $canonicalDocumentName = $this->documentNameResolver->canonicalUrl($renderContext->getDirName(), $node->getTargetReference());
+        $canonicalDocumentName = rtrim($canonicalDocumentName, '.' . $renderContext->getOutputFormat());
```

Comment:
Hm, should this not be done on the parser side? Remove the extension from the file, as we have the extension there? This seems to be a complete list now, but it won't when somebody add's a custom format. Which is supported by the library we are creating.

Reply:
I'll give it another look, that might be a good idea